types/await.php

```diff
@@ -0,0 +1,10 @@
+<?php
+
+use function PHPStan\Testing\assertType;
+use function React\Async\await;
+use function React\Promise\resolve;
+
+$passThroughBoolFn = static fn (bool $bool): bool => $bool;
+
+assertType('bool', await(resolve(true)));
+assertType('bool', await(resolve(true)->then($passThroughBoolFn)));
```

Comment:
Are we testing the template types of the `await()` function or the `then()` method here? Should this be moved to promises or should we keep this here for clarity?

Reply:
Removed it as that should all come from promise anyway.